src/open-local-helper.cpp

```diff
@@ -94,6 +95,34 @@ void OpenLocalHelper::sendOpenLocalFileMessage(const char *url)
     g_object_unref (sync_client_);
 }
 
+QUrl OpenLocalHelper::generateLocalFileSeafileUrl(const QString& repo_id, const Account& account, const QString& path)
+{
+    const QString& email = account.username;
+    QUrl url;
+    url.setScheme(kSeafileProtocolScheme);
+    url.setHost(kSeafileProtocolHostOpenFile);
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+    QUrlQuery url_query;
+    url_query.addQueryItem("repo_id",  repo_id);
+    if (!email.isEmpty())
+        url_query.addQueryItem("email",  email);
+    url_query.addQueryItem("path",  path);
+    url.setQuery(url_query);
+#else
+    url.addQueryItem("repo_id",  repo_id);
+    if (!email.isEmpty())
+        url.addQueryItem("email",  email);
+    url.addQueryItem("path",  path);
+#endif
```

Comment:
There is a convenient method [includeQueryParams](https://github.com/haiwen/seafile-client/blob/v4.2.5/src/utils/utils.h#L89) which wraps the difference of qt4 and qt5.